modules/broker-auth/src/main/java/org/wso2/broker/auth/authentication/authenticator/Authenticator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ *   Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ *
+ */
+package org.wso2.broker.auth.authentication.authenticator;
+
+import org.wso2.broker.auth.BrokerAuthConfiguration;
+import org.wso2.broker.auth.BrokerAuthException;
+import org.wso2.broker.common.StartupContext;
+
+/**
+ * Interface represents the authenticator.
+ * This provides ability to extend different authentication strategies for broker auth.
+ * <p>
+ * {@link #authenticate(String, char[])} is used to authenticate user on given username and credentials.
+ */
+public interface Authenticator {
+
+    /**
+     * Initiate authenticator with startup context.
+     *
+     * @param startupContext the startup context provides registered services for authenticator functionality.
+     * @param authenticationConfiguration the authentication configuration
+     */
+    void initialize(StartupContext startupContext,
+                    BrokerAuthConfiguration.AuthenticationConfiguration authenticationConfiguration);
```

Comment:
Isn't it enough to give the startup context? The implementer can get the required configurations using the startup context.

Reply:
yes. I Generic case we should only have start up context.